algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/treatment/HasRadiologicalProgressionFollowingLatestTreatmentLine.kt

```diff
@@ -6,9 +6,34 @@ import com.hartwig.actin.datamodel.PatientRecord
 import com.hartwig.actin.datamodel.algo.Evaluation
 
 //TODO: Implement according to README
-class HasRadiologicalProgressionFollowingLatestTreatmentLine : EvaluationFunction {
+class HasRadiologicalProgressionFollowingLatestTreatmentLine(
+    private val mustBeRadiological: Boolean
+) :
+    EvaluationFunction {
 
     override fun evaluate(record: PatientRecord): Evaluation {
-        return EvaluationFactory.recoverableUndetermined("Radiological progression following latest treatment line undetermined")
+        val undeterminedMessage = "Radiological progression following latest treatment line undetermined"
+
+        val treatmentHistory = record.oncologicalHistory
+        val systemicTreatments =
+            treatmentHistory.filter { SystemicTreatmentAnalyser.treatmentHistoryEntryIsSystemic(it) }
+        val (systemicTreatmentsWithStartDate, systemicTreatmentsWithoutStartDate) = systemicTreatments.partition { it.startYear != null }
+        val lastTreatment = SystemicTreatmentAnalyser.lastSystemicTreatment(systemicTreatmentsWithStartDate)
+        val lastTreatmentResultedInPD =
+            lastTreatment?.let { ProgressiveDiseaseFunctions.treatmentResultedInPD(it) } == true
+
+        return when {
+            lastTreatmentResultedInPD -> {
+                EvaluationFactory.pass("Last systemic treatment resulted in radiological progression.")
+            }
+            // or stop year is provided, and PD assumed to be radiological
+            lastTreatment?.treatmentHistoryDetails?.stopYear != null && mustBeRadiological -> {
```

Comment:
This "mustBeRadiological" argument confuses a bit.  The name of this class would suggest that this rule can only be used when progression can assumed to be radiologically confirmed. Whereas in HasHadPDFollowingSomeSystemicTreatments would seem to support both the case where is can and can't be assumed.  Would be good to check with Charlotte on that.